src/integrationTest/java/uk/gov/hmcts/cmc/claimstore/controllers/ResendStaffNotificationsCoreCaseDataTest.java

```diff
@@ -0,0 +1,296 @@
+import uk.gov.hmcts.cmc.claimstore.BaseGetTest;
+import uk.gov.hmcts.cmc.claimstore.idam.models.GeneratePinResponse;
+import uk.gov.hmcts.cmc.claimstore.idam.models.User;
+import uk.gov.hmcts.cmc.claimstore.idam.models.UserDetails;
+import uk.gov.hmcts.cmc.claimstore.services.notifications.fixtures.SampleUserDetails;
+import uk.gov.hmcts.cmc.email.EmailData;
+import uk.gov.hmcts.reform.sendletter.api.SendLetterApi;
+import uk.gov.hmcts.reform.sendletter.api.SendLetterResponse;
+
+import java.util.Collections;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyMap;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.put;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.listOfCaseDetails;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.listOfCaseDetailsWithCCJ;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.listOfCaseDetailsWithDefResponse;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.listOfCaseDetailsWithDefendant;
+import static uk.gov.hmcts.cmc.claimstore.utils.ResourceLoader.listOfCaseDetailsWithOfferCounterSigned;
+
+@TestPropertySource(
+    properties = {
+        "document_management.api_gateway.url=false",
+        "core_case_data.api.url=http://core-case-data-api"
+    }
+)
+public class ResendStaffNotificationsCoreCaseDataTest extends BaseGetTest {
+
+    private static final String CASE_REFERENCE = "000MC023";
+
+    @MockBean
+    protected SendLetterApi sendLetterApi;
+
+    @Captor
+    private ArgumentCaptor<EmailData> emailDataArgument;
+
+    private static final String PAGE = "1";
+
+    @Before
+    public void setUp() {
+        given(pdfServiceClient.generateFromHtml(any(byte[].class), anyMap()))
+            .willReturn(new byte[]{1, 2, 3, 4});
+        UserDetails userDetails = SampleUserDetails.builder().withRoles("caseworker-cmc").build();
+        User user = new User(AUTHORISATION_TOKEN, userDetails);
+        given(userService.getUserDetails(AUTHORISATION_TOKEN)).willReturn(userDetails);
+        given(userService.authenticateAnonymousCaseWorker()).willReturn(user);
+        given(userService.getUser(AUTHORISATION_TOKEN)).willReturn(user);
+        given(authTokenGenerator.generate()).willReturn(SERVICE_TOKEN);
+    }
+
+    @Test
+    public void shouldRespond404WhenClaimDoesNotExist() throws Exception {
+        String nonExistingClaimReference = "something";
+
+        given(coreCaseDataApi.searchForCaseworker(
```

Comment:
It would be great to extract method for these mocks to increase readability. We could've:

 `givenSearchByReferenceNumberReturns(nonExistingClaimReference, Collections.emptyList())`

and

 `givenSearchByReferenceNumberReturns(CASE_REFERENCE, listOfCaseDetails)`

Reply:
Alternatively we could have:

`givenSearchByReferenceNumberReturnsEmptyList(nonExistingClaimReference)`

and 

`givenSearchByReferenceNumberReturns(listOfCaseDetails)`